latitudesh/terraform-provider-latitudesh
# feat: implement import for provider

#### What does this PR do?
Implements Import functionality for latitude provider.

#### Description of Task to be completed?
Import will work by passing IDs of the resource to be imported.

#### How should this be manually tested?
Write config of the resource to be imported e.g let's import an existing latitude project
```tf
resource "latitudesh_project" "project" {
  name        = "unkown"
  description = "unkown"
  environment = "unkown"
}
```
Run the following command to import the project.
```sh
terraform import latitudesh_project.project <ProjectID>
```
Once imported observe the state file. Run `terraform plan` to inspect the values. Then change the value of the name, description, and environment from “unknown” in config to what is highlighted in the plan output, and run `terraform plan` again. Notice the output. This time the plan does not indicate any replacements.

For nested resources such as ssh_key and user data, run the following command (after writing a valid config)

```sh
terraform import latitudesh_ssh_key.ssh_key <ProjectID:SshKeyID>
terraform import latitudesh_ssh_key.ssh_key 4509:366
```

A successful import looks like this
![image](https://user-images.githubusercontent.com/103584290/229788939-dc395b4e-51ec-48c0-8ef9-0870884fb90b.png)



#### Any background context you want to provide?
https://latitude.atlassian.net/browse/PD-2915

#### What are the relevant GitHub issues (if any)?
Resolves https://github.com/latitudesh/terraform-provider-latitudesh/issues/32
#### Screenshots (if appropriate):

#### Questions:
